### ISSUE TITLE
investigate/consider parsing LLVM's intrinsic definitions

### ISSUE THREAD

**Issue by sunnygleason** [simd-lite/simd-lite#13]
I found 2 crates that can parse LLVM. Given a proper LLVM
parser, maybe we can parse LLVM definitions in an easy way
to get them into Rust?

* https://crates.io/crates/llvm-ir
* https://crates.io/crates/llvm-sys

It looks like LLVM definitions are in these files, I didn't dive
too deep into them (yet)  😄 

* https://github.com/llvm-mirror/llvm/blob/master/include/llvm/IR/IntrinsicsAArch64.td
* https://github.com/llvm-mirror/llvm/blob/master/include/llvm/IR/IntrinsicsARM.td
* https://github.com/llvm-mirror/llvm/blob/master/include/llvm/IR/IntrinsicsX86.td

If this works, it could be kind of cool because then:

1) maybe we don't need to keep as much data in the `.spec` files
2) the definitions are guaranteed to be accessible to Rust (maybe? need to check)

**Comment by sunnygleason** [comment]
another possibility is parsing C preprocessor output:

```
gcc -D __LITTLE_ENDIAN__ -E /usr/lib/clang/6.0.0/include/arm_neon.h
```

**Comment by Licenser** [comment]
Ja, however generating tests would still be nice to have so, I see that as one of the main benefits for generating the code. Perhaps we can find a nice way to integrate both concepts?

**Comment by twilligon** [comment]
sounds like https://lib.rs/crates/llvmint
